### ISSUE TITLE
Feature Request: Not possible to set subscription_id for implicit IAP logging

### ISSUE THREAD

**Issue by jdmunro** [facebook/facebook-android-sdk#687]
## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing feature requests on [GitHub issues](https://github.com/facebook/facebook-android-sdk/issues)
- [x] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I've given my issue the title: `Feature Request: [name of my feature request]`

## Goals

I'm implementing subscriptions in an app and want to be able to track the full lifecycle of subscriptions. This involves being able to use the Marketing API to send `RecurringSubscriptionPayment` and `CancelSubscription` from the server-side. 

## Expected Results

In order to do this we need to be able to pass a `subscription_id` with the implicitly logged IAP app events that the client-side FBSDK sends. For instance, it makes sense to set this automatically to the subscription purchase token because it can be easily accessed again on the backend rather than using some arbitrary ID.

> The subscription_id is used to link events related to the same subscription.

https://developers.facebook.com/docs/marketing-api/server-side-api/subscription-lifecycle-events

Currently we only have the option of manually implementing IAP logging using the `StartTrial` and `Subscribe` App Events: however, this has an additional problem. In order not to duplicate events we have to turn off the automatic/implicit logging: this can only be done globally via the Facebook Analytics settings - this would break all our existing clients which already make use of this.

In summary:

1. There should be a way to set a `subscription_id` to be used with the implicit IAP logging, OR it should be set automatically to the purchase token which can then be used to link subscription lifecycle events with the sever-side Marketing API

2. There ideally should be a way to turn of implicit IAP logging on the client-side, so that a new version of an app can be deployed that implements its own IAP logging using the same events, thus avoiding duplicates or turning it off globally for everyone
